src/scripts/main.js

```diff
@@ -1,3 +1,388 @@
+  const currentTh = event.target.closest('th');
+
+  if (!currentTh) {
+    return;
+  }
+
+  function arraysAreEqual(arr1, arr2) {
+    if (arr1.length !== arr2.length) {
+      return false;
+    }
+
+    return arr1.every((el, index) => el === arr2[index]);
+  }
+
+  if (currentTh.textContent === 'Name') {
+    const currentRows = tbody.querySelectorAll('tr');
+    const currentRowsCopy = [...currentRows];
+
+    currentRows.forEach((row) => {
+      row.remove();
+    });
+
+    const sortedByNames = [...currentRowsCopy].sort((row1, row2) => {
+      const nameTd1 = row1.firstElementChild;
+      const nameTd2 = row2.firstElementChild;
+
+      return nameTd1.textContent.localeCompare(nameTd2.textContent);
+    });
+
+    if (arraysAreEqual(sortedByNames, currentRowsCopy)) {
+      sortedByNames.reverse();
+    }
+
+    sortedByNames.forEach((row) => {
+      tbody.append(row);
+    });
+  }
+
+  if (currentTh.textContent === 'Position') {
+    const currentRows = tbody.querySelectorAll('tr');
+    const currentRowsCopy = [...currentRows];
+
+    currentRows.forEach((row) => {
+      row.remove();
+    });
+
+    const sortedByPositions = [...currentRowsCopy].sort((row1, row2) => {
+      const positionTd1 = row1.children[1];
+      const positionTd2 = row2.children[1];
+
+      return positionTd1.textContent.localeCompare(positionTd2.textContent);
+    });
+
+    if (arraysAreEqual(sortedByPositions, currentRowsCopy)) {
+      sortedByPositions.reverse();
+    }
+
+    sortedByPositions.forEach((row) => {
+      tbody.append(row);
+    });
+  }
+
+  if (currentTh.textContent === 'Office') {
+    const currentRows = tbody.querySelectorAll('tr');
+    const currentRowsCopy = [...currentRows];
+
+    currentRows.forEach((row) => {
+      row.remove();
+    });
+
+    const sortedByOffices = [...currentRowsCopy].sort((row1, row2) => {
+      const officeTd1 = row1.children[2];
+      const officeTd2 = row2.children[2];
+
+      return officeTd1.textContent.localeCompare(officeTd2.textContent);
+    });
+
+    if (arraysAreEqual(sortedByOffices, currentRowsCopy)) {
+      sortedByOffices.reverse();
+    }
+
+    sortedByOffices.forEach((row) => {
+      tbody.append(row);
+    });
+  }
+
+  if (currentTh.textContent === 'Age') {
+    const currentRows = tbody.querySelectorAll('tr');
+    const currentRowsCopy = [...currentRows];
+
+    currentRows.forEach((row) => {
+      row.remove();
+    });
+
+    const sortedByAges = [...currentRowsCopy].sort((row1, row2) => {
+      const ageTd1 = row1.children[3];
+      const ageTd2 = row2.children[3];
+
+      return Number(ageTd2.textContent) - Number(ageTd1.textContent);
+    });
+
+    if (arraysAreEqual(sortedByAges, currentRowsCopy)) {
+      sortedByAges.reverse();
+    }
+
+    sortedByAges.forEach((row) => {
+      tbody.append(row);
+    });
+  }
+
+  function SToNumber(string) {
+    const filtredString = string.replace(/\$/g, '').replace(/,/g, '');
+
+    return Number(filtredString);
```

Comment:
The `SToNumber` function should handle cases where the input string is not a valid number after removing dollar signs and commas. Consider adding a check to return NaN or throw an error if the conversion fails.